### ISSUE TITLE
bazel crashes when run with incorrect flags in envoy

### ISSUE THREAD

**Issue by janakdr** [bazelbuild/bazel#10873]
This is with Bazel 2.0.0. Repro instructions:

```shell
git clone https://github.com/tetratelabs/getenvoy-package
cd getenvoy-package/envoy_pkg
bazel build --config=remote
```

You may need to have authenticated with RBE in some way, but I don't think so. The crash is:

```
$ bazel --bazelrc=/dev/null build --config=remote
Starting local Bazel server and connecting to it...
WARNING: Usage: bazel build <options> <targets>.
Invoke `bazel help build` for full description of usage and options.
Your request is correct, but requested an empty set of targets. Nothing will be built.
DEBUG: /usr/local/google/home/janakr/.cache/bazel/_bazel_janakr/233251a613eddab5a2c855ce3c799438/external/bazel_toolchains/rules/rbe_repo/checked_in.bzl:113:13: rbe_linux_glibc not using checked in configs; registry was set to 'gcr.io' and toolchain_config_suite_spec is configured for 'marketplace.gcr.io'
INFO: Analyzed 0 targets (0 packages loaded, 0 targets configured).
Internal error thrown during build. Printing stack trace: java.lang.IllegalStateException: actionContextProvider was null
	at com.google.common.base.Preconditions.checkState(Preconditions.java:507)
	at com.google.devtools.build.lib.remote.RemoteModule.afterAnalysis(RemoteModule.java:371)
	at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.execute(AnalysisPhaseRunner.java:128)
	at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:143)
	at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:271)
	at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:99)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:543)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:205)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:601)
	at com.google.devtools.build.lib.server.GrpcServerImpl.lambda$run$2(GrpcServerImpl.java:652)
	at io.grpc.Context$1.run(Context.java:595)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

INFO: Elapsed time: 2.035s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
Internal error thrown during build. Printing stack trace: java.lang.IllegalStateException: actionContextProvider was null
	at com.google.common.base.Preconditions.checkState(Preconditions.java:507)
	at com.google.devtools.build.lib.remote.RemoteModule.afterAnalysis(RemoteModule.java:371)
	at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.execute(AnalysisPhaseRunner.java:128)
	at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:143)
	at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:271)
	at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:99)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:543)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:205)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:601)
	at com.google.devtools.build.lib.server.GrpcServerImpl.lambda$run$2(GrpcServerImpl.java:652)
	at io.grpc.Context$1.run(Context.java:595)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
java.lang.IllegalStateException: actionContextProvider was null
	at com.google.common.base.Preconditions.checkState(Preconditions.java:507)
	at com.google.devtools.build.lib.remote.RemoteModule.afterAnalysis(RemoteModule.java:371)
	at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.execute(AnalysisPhaseRunner.java:128)
	at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:143)
	at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:271)
	at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:99)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:543)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:205)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:601)
	at com.google.devtools.build.lib.server.GrpcServerImpl.lambda$run$2(GrpcServerImpl.java:652)
	at io.grpc.Context$1.run(Context.java:595)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
FAILED: Build did NOT complete successfully
```
It was pointed out to me that adding the --remote_executor flag avoids the crash.

**Comment by pcj** [comment]
```
FAILED: Build did NOT complete successfully
Internal error thrown during build. Printing stack trace: java.lang.IllegalStateException: actionContextProvider was null
	at com.google.common.base.Preconditions.checkState(Preconditions.java:507)
	at com.google.devtools.build.lib.remote.RemoteModule.afterAnalysis(RemoteModule.java:373)
	at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.execute(AnalysisPhaseRunner.java:128)
	at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:142)
	at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:270)
	at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:99)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:543)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:205)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:601)
	at com.google.devtools.build.lib.server.GrpcServerImpl.lambda$run$2(GrpcServerImpl.java:652)
	at io.grpc.Context$1.run(Context.java:595)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
java.lang.IllegalStateException: actionContextProvider was null
	at com.google.common.base.Preconditions.checkState(Preconditions.java:507)
	at com.google.devtools.build.lib.remote.RemoteModule.afterAnalysis(RemoteModule.java:373)
	at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.execute(AnalysisPhaseRunner.java:128)
	at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:142)
	at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:270)
	at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:99)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:543)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:205)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:601)
	at com.google.devtools.build.lib.server.GrpcServerImpl.lambda$run$2(GrpcServerImpl.java:652)
	at io.grpc.Context$1.run(Context.java:595)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
FAILED: Build did NOT complete successfully
```

Also observed on bazel 2.1.1

**Comment by coeuvre** [comment]
Should be fixed by #11898. Closing.
